### PR TITLE
Feat/switch typescript

### PR DESCRIPTION
**Pull request by EdwardIrby**
switch refactor

### CODE REVIEW

File: packages/switch/src/react/__stories__/index.story.tsx [hunk 1]
```diff
@@ -43,64 +42,71 @@ storiesOf('click', module)
...
+  Object.values(Switch.labelAligns).forEach(labelAlign =>
+    labelStory.add(`${size} ${labelAlign}`, () => (
       <Switch size={size} labelAlign={labelAlign}>
         Click me
       </Switch>
     ))
   )
 )
 
+type RenderProp = {
```
Comment (jaketrent) [inline packages/switch/src/react/__stories__/index.story.tsx:55]: nit: It's test, perhaps less important...

"RenderProp" seems to misname what this is.  The render prop is the function given to the parent component.  These are more like "RenderPropProps".

File: packages/switch/src/react/index.tsx [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+
+const styles = {
+  switch: (disabled: boolean, labelAlign: ValueOf<typeof vars.labelAligns>) =>
+    compose(
+      css(stylesheet['.psds-switch']),
+      css(stylesheet[`.psds-switch--labelAlign-${labelAlign}`]),
+      disabled && css(stylesheet['.psds-switch--disabled'])
+    ),
+
+  track: (props: {
```
Comment (jaketrent) [inline packages/switch/src/react/index.tsx:26]: I'm fine with not passing props to all these.  But there are functions with 3-4 params, and redefined SwitchProps.  I'd consider not defining these props in multiple places for this many.  

Something like:

```
(props: Pick<SwitchProps, 'checked', 'color', 'size'>) => ...
```

Seems DRYer.

File: packages/switch/src/react/index.tsx [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+}
+
+interface SwitchProps
+  extends Omit<HTMLAttributes<HTMLInputElement | HTMLLabelElement>, 'onClick'> {
+  checked?: boolean
+  color?: ValueOf<typeof vars.colors>
+  disabled?: boolean
+  error?: boolean
+  labelAlign?: ValueOf<typeof vars.labelAligns>
+  onBlur?: (evt: React.FocusEvent) => void
```
Comment (jaketrent) [inline packages/switch/src/react/index.tsx:90]:
```suggestion
  onBlur?: React.FocusEventHandler
```

File: packages/switch/src/react/index.tsx [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+  onClick?: (evt: boolean) => void
+  onFocus?: (evt: React.FocusEvent) => void
```
Comment (jaketrent) [inline packages/switch/src/react/index.tsx:92]:
```suggestion
  onFocus?: React.FocusEventHandler
```

File: packages/switch/src/react/index.tsx [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+
+interface SwitchProps
+  extends Omit<HTMLAttributes<HTMLInputElement | HTMLLabelElement>, 'onClick'> {
+  checked?: boolean
+  color?: ValueOf<typeof vars.colors>
+  disabled?: boolean
+  error?: boolean
+  labelAlign?: ValueOf<typeof vars.labelAligns>
+  onBlur?: (evt: React.FocusEvent) => void
+  onClick?: (evt: boolean) => void
```
Comment (jaketrent) [inline packages/switch/src/react/index.tsx:91]:
```suggestion
  onClick?: (checked: boolean) => void
```

**Review by jaketrent — CHANGES_REQUESTED** [review]
Mostly looks good.

Still getting this on tsc:

```
design-system/packages/switch on  feat/switch-typescript [$] ❯ tsc
src/react/index.tsx:125:51 - error TS2322: Type 'HTMLInputElement | undefined' is not assignable to type 'HTMLInputElement'.
  Type 'undefined' is not assignable to type 'HTMLInputElement'.

125     React.useImperativeHandle(forwardedRef, () => ref.current)
                                                      ~~~~~~~~~~~

  ../../node_modules/@types/react/index.d.ts:1093:79
    1093     function useImperativeHandle<T, R extends T>(ref: Ref<T>|undefined, init: () => R, deps?: DependencyList): void;
                                                                                       ~~~~~~~
    The expected type comes from the return type of this signature.


Found 1 error.
```

**Comment by EdwardIrby** [comment]
> Mostly looks good.
> 
> Still getting this on tsc:
> 
> ```
> design-system/packages/switch on  feat/switch-typescript [$] ❯ tsc
> src/react/index.tsx:125:51 - error TS2322: Type 'HTMLInputElement | undefined' is not assignable to type 'HTMLInputElement'.
>   Type 'undefined' is not assignable to type 'HTMLInputElement'.
> 
> 125     React.useImperativeHandle(forwardedRef, () => ref.current)
>                                                       ~~~~~~~~~~~
> 
>   ../../node_modules/@types/react/index.d.ts:1093:79
>     1093     function useImperativeHandle<T, R extends T>(ref: Ref<T>|undefined, init: () => R, deps?: DependencyList): void;
>                                                                                        ~~~~~~~
>     The expected type comes from the return type of this signature.
> 
> 
> Found 1 error.
> ```

lint:fix seems to be removing some of my typecasting, weird?

File: packages/switch/src/react/index.tsx (hunk 2 above)
Comment (EdwardIrby) [reply packages/switch/src/react/index.tsx:26]: Look like that won't work

**Review by jaketrent — APPROVED** [review]

**Pull request merged**